recipes/default.rb

```diff
@@ -10,26 +10,14 @@
   compile_time false if respond_to?(:compile_time)
 end
 
-if platform_family? 'rhel'
-  yum_repository 'influxdb' do
-    description 'InfluxDB Repository - RHEL \$releasever'
-    baseurl node['influxdb']['upstream_repository']
-    gpgkey 'https://repos.influxdata.com/influxdb.key'
+influxdb_install 'influxdb' do
+  if node['kernel']['machine'].include? '64'
+    arch_type 'amd64'
+  elsif node['kernel']['machine'].include? '386'
+    arch_type 'i386'
+  else
+    Chef::Log.warning "I am unable to determine arch type for: #{node['kernel']['machine']}"
   end
```

Comment:
As an abstraction for installing the package influxdb, I would prefer if all the logic can be inside the resource.  the `package` and yum_repository and apt_repository or any of the repository should detect the machine right?